src/main/java/com/minecolonies/coremod/commands/CommandEntryPointNew.java

```diff
@@ -0,0 +1,596 @@
+            this.executionTreeNode = executionTreeNode;
+            this.executionActionArgumentList = executionActionArgumentList;
+            this.badArgument = badArgument;
+        }
+
+        public List<String> getTabCompletions()
+        {
+            return tabCompletions;
+        }
+
+        public TreeNode<Menu> getExecutionTreeNode()
+        {
+            return executionTreeNode;
+        }
+
+        public List<ActionArgument> getExecutionActionArgumentList()
+        {
+            return executionActionArgumentList;
+        }
+
+        public String getBadArgument()
+        {
+            return badArgument;
+        }
+
+        public void setBadArgument(final String badArgument)
+        {
+            this.badArgument = badArgument;
+        }
+
+    }
+
+    public static final class ActionMenuHolder
+    {
+        @NotNull private final TreeNode<Menu> treeNode;
+        @NotNull private final ActionArgument actionArgument;
+        private Object value;
+
+        ActionMenuHolder(@NotNull final TreeNode<Menu> treeNode, @NotNull final ActionArgument actionArgument)
+        {
+            super();
+            this.treeNode = treeNode;
+            this.actionArgument = actionArgument;
+        }
+        public TreeNode<Menu> getTreeNode()
+        {
+            return treeNode;
+        }
+        public ActionArgument getActionArgument()
+        {
+            return actionArgument;
+        }
+        public Object getValue()
+        {
+            return value;
+        }
+        public void setValue(final Object value)
+        {
+            this.value = value;
+        }
+    }
+
+    /**
+     * The level required to execute /mc commands.
+     * private static final int OP_PERMISSION_LEVEL = 3;
+     */
+
+    /**
+     * Create our entry point once.
+     */
+    public CommandEntryPointNew()
+    {
+        super();
+        root = buildMenu(NavigationMenuType.MINECOLONIES);
+    }
+
+    /**
+     * Removed so we can control this at the config
+     * Only allow OP's execute the commands.
+     *
+     * @return the int permission level (3 for OP).
+     * public int getRequiredPermissionLevel(){return OP_PERMISSION_LEVEL;}
+     */
+    @NotNull
+    @Override
+    public List<String> getAliases()
+    {
+        return Arrays.asList("mc", "col", "mcol", "mcolonies", "minecol", "minecolonies");
+    }
+
+    private TreeNode<Menu> buildMenu(@NotNull final MenuType menuType)
+    {
+        return buildMenu(menuType, new HashSet<>());
+    }
+
+    private TreeNode<Menu> buildMenu(@NotNull final MenuType menuType, @NotNull final Set<MenuType> menuTypesSoFar)
+    {
+        // Prevent recursion
+        if (menuTypesSoFar.contains(menuType))
+        {
+            return null;
+        }
+        else
+        {
+            menuTypesSoFar.add(menuType);
+        }
+
+        final Menu menu = menuType.getMenu();
+        final TreeNode<Menu> treeNode = new TreeNode<Menu>(menu);
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace the type specification in this constructor call with the diamond operator ("<>"). (sonar.java.source not set. Assuming 7 or greater.) [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2293)

Reply:
this